### PR TITLE
Add support for callback header

### PR DESCRIPTION
**Pull request by bergundy**
Exposed callback headers as defined in this recent change to the API:

https://github.com/nexus-rpc/api/pull/4

### CODE REVIEW

File: nexus/client.go [hunk 1]
```diff
@@ -259,6 +260,9 @@ type ExecuteOperationOptions struct {
 	// Even though Client.ExecuteOperation waits for operation completion, some applications may want to set this
 	// callback as a fallback mechanism.
 	CallbackURL string
+	// Optional header fields set by a client that are required to be attached to the callback request when an
+	// asynchronous operation completes.
+	CallbackHeader Header
```
Comment (MichaelSnowden) [inline nexus/client.go:265]: If there can only be one value per key here, consider just using a map.

**Review by MichaelSnowden — APPROVED** [review]

File: nexus/api.go [hunk 2]
```diff
@@ -117,9 +117,22 @@ func (h Header) Get(k string) string {
...
+	}
+	return header
+}
+
+func httpHeaderToCallbackHeader(httpHeader http.Header) Header {
+	header := Header{}
+	for k, v := range httpHeader {
+		lowerK := strings.ToLower(k)
+		if strings.HasPrefix(lowerK, "nexus-callback-") {
+			// Nexus headers can only have single values, ignore multiple values.
```
[the remark this answers is not in the document]
Comment (bergundy) [reply nexus/api.go:134]: For compatibility with gRPC, Kafka, and Temporal headers.
This is the lowest common denominator.

File: nexus/api.go [hunk 3]
```diff
@@ -117,9 +117,22 @@ func (h Header) Get(k string) string {
...
+			header[lowerK[15:]] = v[0]
```
[the remark this answers is not in the document]
Comment (bergundy) [reply nexus/api.go:135]: Sure, but it really doesn't matter.

File: nexus/client.go (hunk 1 above)
Comment (bergundy) [reply nexus/client.go:265]: Header has a case insensitive getter, it's convenient.

File: nexus/api.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (bergundy) [reply nexus/api.go:134]: I thought I put it in the spec. I'll update, thanks.

**Pull request merged**